include/flatbuffers/code_generators.h

```diff
@@ -95,6 +95,10 @@ class BaseGenerator {
 
   static std::string NamespaceDir(const Parser &parser, const std::string &path,
                                   const Namespace &ns);
+  static std::string NamespaceDir(const Parser &parser, const std::string &path,
+                                  const Namespace &ns, const bool dasherize);
+
```

Comment:
It is not necessary to have two distinct methods for `NamespaceDir`. You can declare a default value for the latest argument:
```cpp
static std::string NamespaceDir(const Parser &parser, const std::string &path,
                                  const Namespace &ns, const bool dasherize = false);
```
The same for the `NamespaceDir(const Namespace &ns, const bool dasherize)`.